src/RequestInformation.cs

```diff
@@ -174,27 +172,27 @@ public void AddQueryParameters<T>(T source)
             }
         }
 
-        private static object ExpandArray(Array collection)
+        private static string[] ExpandArray(Array collection)
         {
             var passedArray = new string[collection.Length];
             for(var i = 0; i < collection.Length; i++)
             {
-                passedArray[i] = GetSanitizedValue(collection.GetValue(i)!).ToString()!;
+                passedArray[i] = GetSanitizedValue(collection.GetValue(i)!)!;
             }
             return passedArray;
         }
 
-        private static object ReplaceEnumValueByStringRepresentation(object source)
+        private static string ReplaceEnumValueByStringRepresentation(object source)
         {
             if(source is Enum enumValue && GetEnumName(enumValue) is string enumValueName)
             {
                 return enumValueName;
             }
             
-            return source;
+            return source.ToString()!;
```

Comment:
this and the return type need to be reverted as a bunch of number types (and other cases) will serialize abnormally because of it.

https://github.com/std-uritemplate/std-uritemplate/blob/3c908404bdc8f05e6f04210c3eeb04fffb31c8ac/csharp/Std.UriTemplate.cs#L472

Reply:
I believe the toString was called in the expand array method because when we switched to std uri template, the library wasn't properly processing arrays of anything else than strings initially. But I believe this has been resolved at this point and the call to ToString there should be removed for the reasons outlined in my previous comment. @andrueastman can you confirm?
The alternative being to add the missing types in the GetSanitizedValue method

Reply:
What @baywet mentions is correct. A different library was used before for the Uri template matching before switching to std uri template which handled things differently causing a few issues due to different support levels(e.g handling collections/Guids and some of the custom types in this lib). 
At the end of the day, I think the std library should get an object and return a correct url based on what's right in the spec(in an ideal world, the `GetSanitizedValue` should get smaller and cease to exist).

Reply:
Ok, @andrueastman -  What change should I employ at this point, all things considered?

Reply:
Okay with reverting